File: check_cb.py
import re
import os

callback_regex = re.compile(r'callback_data=["\']([^"\']+)["\']')
handler_regex = re.compile(r'@router\.callback_query\([F\.]*data == ["\']([^"\']+)["\']\)')

defined_callbacks = set()
handled_callbacks = set()

for root, _, files in os.walk("telegram_bot"):
    for file in files:
        if not file.endswith(".py"): continue
        path = os.path.join(root, file)
        with open(path, "r", encoding="utf-8") as f:
            content = f.read()
            for m in callback_regex.findall(content):
                defined_callbacks.add(m)
            for m in handler_regex.findall(content):
                handled_callbacks.add(m)

print("DEFINED but not HANDLED:")
for cb in defined_callbacks - handled_callbacks:
    # Some might be dynamic or prefix based, but let's see
    print(cb)

print("\nHANDLED but not DEFINED:")
for cb in handled_callbacks - defined_callbacks:
    print(cb)

File: check_cb.js
const fs = require('fs');
const path = require('path');

const cbRegex = /callback_data=["']([^"']+)["']/g;
const handlerRegex = /@router\.callback_query\([F\.]*data\s*==\s*["']([^"']+)["']\)/g;

const defined = new Set();
const handled = new Set();

function walk(dir) {
    for (const file of fs.readdirSync(dir)) {
        const full = path.join(dir, file);
        if (fs.statSync(full).isDirectory()) {
            walk(full);
        } else if (file.endsWith('.py')) {
            const content = fs.readFileSync(full, 'utf-8');
            let m;
            while ((m = cbRegex.exec(content)) !== null) defined.add(m[1]);
            while ((m = handlerRegex.exec(content)) !== null) handled.add(m[1]);
        }
    }
}

walk('telegram_bot');

console.log("DEFINED but not HANDLED:");
for (const cb of [...defined].filter(x => !handled.has(x))) {
    console.log(cb);
}

console.log("\nHANDLED but not DEFINED:");
for (const cb of [...handled].filter(x => !defined.has(x))) {
    console.log(cb);
}
